src/build_queue.rs

```diff
@@ -726,9 +729,75 @@ pub async fn queue_rebuilds(
     Ok(())
 }
 
+/// Queue rebuilds for failed crates due to a faulty version of rustdoc
+///
+/// It is assumed that the version of rustdoc matches the one of rustc, which is persisted in the DB.
+/// The priority of the resulting rebuild requests will be lower than previously failed builds.
+/// If a crate is already queued to be rebuilt, it will not be requeued.
+/// Start date is inclusive, end date is exclusive.
+#[instrument(skip_all)]
+pub async fn queue_rebuilds_faulty_rustdoc(
+    conn: &mut sqlx::PgConnection,
+    build_queue: &AsyncBuildQueue,
+    start_nightly_date: &NaiveDate,
+    end_nightly_date: &Option<NaiveDate>,
+) -> Result<i32> {
+    let end_nightly_date =
+        end_nightly_date.unwrap_or_else(|| start_nightly_date.succ_opt().unwrap());
+    let mut results = sqlx::query!(
+        r#"
+SELECT c.name,
+       r.version AS "version: Version"
+FROM crates AS c
+         JOIN releases AS r
+              ON c.latest_version_id = r.id
+                  AND r.rustdoc_status = TRUE
+         JOIN LATERAL (
```

Comment:
you can join `release_build_status` instead, which should have an aggregated, updated, build status for that release.

Reply:
I see, I missed that, thank you! I simplified the query by joining on `release_build_status`, although I had to join on `builds` to be able to filter by the input rustdoc nightly. Since the `release_build_status` table does not have a key pointing at the original build record, I had to join based on the `last_build_time` timestamp being equal to `build.build_finished`, which I assume it is acceptable since `last_build_time` is calculated as the maximum build_finished timestamp across all builds for a specific release.